chain/byteexec/byteexec.go

```diff
@@ -47,16 +47,23 @@ func RunOnLocal(executable []byte, timeOut time.Duration, args ...string) ([]byt
 }
 
 // RunOnDocker runs the given executable in a new docker container.
-func RunOnDocker(executable []byte, timeOut time.Duration, args ...string) ([]byte, error) {
+func RunOnDocker(executable []byte, sandboxMode bool, timeOut time.Duration, args ...string) ([]byte, error) {
 	dir, filename, err := writeFile(executable)
 	if err != nil {
 		return nil, err
 	}
 	defer os.RemoveAll(dir) // clean up
 
-	rawID, err := exec.Command(
-		"docker", "run", "-d", "--rm", "band-provider", "sleep", fmt.Sprintf("%d", int(timeOut.Seconds())),
-	).Output()
+	commands := []string{
+		"run", "-d", "--rm", "band-provider", "sleep", fmt.Sprintf("%d", int(timeOut.Seconds())),
+	}
+	if sandboxMode {
```

Comment:
This looks a bit confusing. Why don't we do.
```
commands := []string{"run"}
if sandboxMode {
   ...
}
commands = append(commands, "-d", "--rm", ...)
```

Reply:
You got it!